google/resource_storage_default_object_acl.go

```diff
@@ -86,13 +77,17 @@ func resourceStorageDefaultObjectAclRead(d *schema.ResourceData, meta interface{
 	for _, v := range res.Items {
 		role := v.Role
 		entity := v.Entity
-		// We only store updates to the locally defined access controls
-		if _, in := reLocalMap[entity]; in {
-			roleEntities = append(roleEntities, fmt.Sprintf("%s:%s", role, entity))
-			log.Printf("[DEBUG]: saving re %s-%s", v.Role, v.Entity)
+
+		//skip project level permissions for the object
```

Comment:
could you please elaborate on what's happening here?

Reply:
Previously we only cared for local ACLs(defined in state file), and ignored the rest. That approach  couldn't really work for import , since there is nothing in the state file to check against. Besides,  if somebody added a permission manually, that wasn't in tf file, terraform state file would never trigger a diff, because it only cared for what's defined in the config.

When the bucket gets created, it gets a default project level set of permissions like this: 
```
{
    "entity": "project-owners-601463989707",
    "projectTeam": {
      "projectNumber": "601463989707",
      "team": "owners"
    },
    "role": "OWNER"
  },
```
I'm checking for the ProjectTeam property and skipping project level permissions (since we don't control them),and import everything else on every read.

Reply:
Do these records get create when we create a bucket via terraform? I remember there was a debate about it and if I recall correctly, you can't remove some of these "OWNER" records without replacing them with something else first.

This change is major and may create unexpected results for other people. Like when somebody explicitly granted project level permissions with terraform. We should find a way to import what we need in a spirit of its current behavior - care about what's defined. It's not great but changing this logic is a major change